tests/stratagus/test_action_built.cpp

```diff
@@ -0,0 +1,251 @@
+//       _________ __                 __
+//      /   _____//  |_____________ _/  |______     ____  __ __  ______
+//      \_____  \\   __\_  __ \__  \\   __\__  \   / ___\|  |  \/  ___/
+//      /        \|  |  |  | \// __ \|  |  / __ \_/ /_/  >  |  /\___ |
+//     /_______  /|__|  |__|  (____  /__| (____  /\___  /|____//____  >
+//             \/                  \/          \//_____/            \/
+//  ______________________                           ______________________
+//                        T H E   W A R   B E G I N S
+//         Stratagus - A free fantasy real time strategy game engine
+//
+/**@name test_action_built.cpp - The test file for action_built.cpp. */
+//
+//      (c) Copyright 2024 by Matthias Schwarzott
+//
+//      This program is free software; you can redistribute it and/or modify
+//      it under the terms of the GNU General Public License as published by
+//      the Free Software Foundation; only version 2 of the License.
+//
+//      This program is distributed in the hope that it will be useful,
+//      but WITHOUT ANY WARRANTY; without even the implied warranty of
+//      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//      GNU General Public License for more details.
+//
+//      You should have received a copy of the GNU General Public License
+//      along with this program; if not, write to the Free Software
+//      Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+//      02111-1307, USA.
+//
+
+#include <doctest.h>
+
+#include "stratagus.h"
+#include "unit.h"
+#include "action/action_built.h"
+#include "construct.h"
+
+class tested_COrder_Built : public COrder_Built
+{
+public:
+	using COrder_Built::ProgressCounter;
+
+	int testUpdateConstrFrame_Percent(CUnit &unit, int percent, int initialFrame = 9999)
+	{
+		const int progressOnePercent = unit.Type->Stats[0].Costs[TimeCost] * 6;
+		return testUpdateConstrFrame_Progress(unit, percent * progressOnePercent, initialFrame);
+		}
+
+	int testUpdateConstrFrame_Progress(CUnit &unit, int progress, int initialFrame = 9999)
+	{
+		unit.Frame = initialFrame; // to see if it was set
+		Frame = -5; // force re-calculation
+		ProgressCounter = progress;
+		UpdateConstructionFrame(unit);
+		return unit.Frame;
+	}
+};
+
+namespace {
+	void setFrames(std::vector<CConstructionFrame> &cframes,
+				   const std::vector<std::pair<int, int>>& vec)
```

Comment:
Would `std::vector<CConstructionFrame> makeFrames(const std::vector<std::pair<int, int>>& vec)` make sense instead?

Reply:
changed. But does not look too different.